plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/OptInPopup.java

```diff
@@ -0,0 +1,105 @@
+package com.google.cloud.tools.eclipse.usagetracker;
+
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.graphics.Font;
+import org.eclipse.swt.graphics.FontData;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.graphics.Rectangle;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Link;
+import org.eclipse.swt.widgets.Shell;
+
+/**
+ * A one-time dialog to suggest opt-in for sending client-side usage metrics.
+ */
+class OptInDialog extends Dialog {
+
+  public OptInDialog(Shell parentShell) {
+    super(parentShell);
+    setShellStyle(SWT.MODELESS);
+    setBlockOnOpen(false);
+  }
+
+  /**
+   * Show this dialog at the top-right corner.
+   */
+  @Override
+  protected Point getInitialLocation(Point initialSize) {
+    Rectangle parentBounds = getParentShell().getBounds();
+    Rectangle parentClientArea = getParentShell().getClientArea();
+
+    int heightCaptionAndUpperBorder =
+        parentBounds.height - parentClientArea.height - getParentShell().getBorderWidth();
+    return new Point(parentBounds.x + parentClientArea.width - initialSize.x,
+        parentBounds.y + heightCaptionAndUpperBorder);
+  }
+
+  /**
+   * Overridden in order to remove the button bar area (and the OK and CANCEL buttons) that the
```

Comment:
Sounds good. We can include @patflynn in the discussion. FYI, below is what we do in IntelliJ.

![intellij-opt-in](https://cloud.githubusercontent.com/assets/10523105/16046095/3a07d4ac-3219-11e6-950c-0a23d2e9e477.png)

Also a corresponding entry in settings:

![intellij-opt-in-settings](https://cloud.githubusercontent.com/assets/10523105/16046131/533d884a-3219-11e6-9c89-ed15cdd00117.png)